### PR TITLE
Remove option to scale grain size in lower mantle

### PR DESCRIPTION
**Pull request by jdannberg**
This option was only useful because you would get large jumps in the grain size compositional field between upper and lower mantle, but now we can use particles instead. 

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ ] I have tested my new feature locally to ensure it is correct.
* [ ] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: source/material_model/grain_size.cc [hunk 1]
```diff
@@ -1457,22 +1449,23 @@ namespace aspect
...
           for (unsigned int i=0; i<dislocation_creep_prefactor.size(); ++i)
             dislocation_creep_prefactor[i] = std::pow(dislocation_creep_prefactor[i],-1.0/dislocation_creep_exponent[i]);
 
-          if (grain_size_evolution_formulation == Formulation::paleowattmeter)
-            boundary_area_change_work_fraction[boundary_area_change_work_fraction.size()-1] /= pv_grain_size_scaling;
+          const double pv_grain_size_scaling         = prm.get_double ("Lower mantle grain size scaling");
+          AssertThrow(pv_grain_size_scaling == 1.0,
+                      ExcMessage("Error: The 'Lower mantle grain size scaling' parameter "
+                                 "has been removed. Please remove it from your input file. For models "
+                                 "with large sptial variations in grain size, please advect your "
```
Comment (gassmoeller) [inline source/material_model/grain_size.cc:1467]: Here and for the other message:

```suggestion
                                 "with large spatial variations in grain size, please advect your "
```

**Review by gassmoeller — COMMENTED** [review]
One typo

**Comment by jdannberg** [comment]
Thank you! I've addressed your comment.

**Review by gassmoeller — APPROVED** [review]
:+1:

**Pull request merged**